Math2/ShapeUtils.cs

```diff
@@ -1,37 +1,147 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
 
 namespace SharpMath2
 {
+    /// <summary>
+    /// A simple shape utils class.
+    /// </summary>
     public class ShapeUtils
     {
+        /// <summary>
+        /// A dictionary containing the circle shapes.
+        /// </summary>
+        private static Dictionary<Tuple<float, float, float, float>, Polygon2> CircleCache = new Dictionary<Tuple<float, float, float, float>, Polygon2>();
+
+        /// <summary>
+        /// A dictionary containing the rectangle shapes.
+        /// </summary>
+        private static Dictionary<Tuple<float, float, float, float>, Polygon2> RectangleCache = new Dictionary<Tuple<float, float, float, float>, Polygon2>();
+
+        /// <summary>
+        /// A dictionary containing the convex polygon shapes.
+        /// </summary>
+        private static Dictionary<int, Polygon2> ConvexPolygonCache = new Dictionary<int, Polygon2>();
+
+        /// <summary>
+        /// Creates a convex polygon from the given texture.
+        /// </summary>
+        /// <param name="Texture"></param>
+        public static Polygon2 CreateConvexPolygon(Texture2D Texture)
+        {
+            var Key = Texture.GetHashCode();
+
+            if (ConvexPolygonCache.ContainsKey(Key))
+                return ConvexPolygonCache[Key];
+
+            var data = new uint[Texture.Width * Texture.Height];
+            Texture.GetData<uint>(data);
+
+            var Points = new List<Vector2>();
+
+            for (var i = 0; i < Texture.Width; i++)
+            for (var j = 0; j < Texture.Height; j++)
+                if (data[j * Texture.Width + i] != 0)
+                    Points.Add(new Vector2(i, j));
+
+            if (Points.Count <= 2)
+                throw new Exception("Can not create a convex hull from a line.");
+
+            int n = Points.Count, k = 0;
+            List<Vector2> H = new List<Vector2>(new Vector2[2 * n]);
+
+            Points.Sort((a, b) => a.X == b.X ? a.Y.CompareTo(b.Y) : (a.X > b.X ? 1 : -1));
```

Comment:
I believe you can make both sides of the ternary the same (either using an inner ternary or make the right side a.X.CompareTo(b.X) - minor readability change